File: remote/src/main/java/com/kelsos/mbrc/presenters/AlbumTracksPresenter.java
package com.kelsos.mbrc.presenters;

import com.kelsos.mbrc.annotations.Queue;
import com.kelsos.mbrc.domain.Track;
import com.kelsos.mbrc.ui.views.AlbumTrackView;

public interface AlbumTracksPresenter {
  void bind(AlbumTrackView view);

  void load(long albumId);

  void play(long albumId);

  void queue(Track entry, @Queue.Action String action);
}

File: remote/src/main/kotlin/com/kelsos/mbrc/presenters/AlbumTracksPresenter.kt
package com.kelsos.mbrc.presenters

import com.kelsos.mbrc.annotations.Queue
import com.kelsos.mbrc.domain.Track
import com.kelsos.mbrc.ui.views.AlbumTrackView

interface AlbumTracksPresenter {
  fun bind(view: AlbumTrackView)

  fun load(albumId: Long)

  fun play(albumId: Long)

  fun queue(entry: Track, @Queue.Action action: String)
}
